book/frontend/tutorial/step_7.md

```diff
@@ -8,32 +8,56 @@ In this step we will improve the UI of our beloved `Hello` component using the b
 
 Let's perform the same steps that we did previously for the loading spinner, but this time adding some customization on top of the standard b-r-c component:
 
-1. Create a `src/app/components/Basic/Panel/Panel.js` file as follows:
-```js
-import React from 'react';
-import BRCPanel from 'buildo-react-components/lib/Panel/Panel';
-import skinnable from 'react-skinnable';
+Create a `src/app/components/Basic/Panel/Panel.tsx` file as follows:
+
+```tsx
+import * as React from 'react';
+import { default as BRCPanel, PanelProps } from 'buildo-react-components/lib/Panel/Panel';
+import { ObjectOmit } from 'typelevel-ts';
 
 import './panel.scss';
 
-@skinnable()
-export default class Panel extends React.PureComponent {
+type DefaultProps = {
+  type: PanelProps['type']
+};
+
+type RequiredProps = ObjectOmit<PanelProps, keyof DefaultProps>;
+
+type Props = Partial<DefaultProps> & RequiredProps;
+
+const defaultProps: DefaultProps = {
+  type: 'docked-top'
+};
```

Comment:
I'd avoid `ObjectOmit` entirely here (unnecessary complexity for this tutorial), and instead default props following this strategy: https://paper.dropbox.com/doc/TS-DefaultProps-AVtary8J9rJd72ZykLsk2
(this should be updated in the webseed tutorial branch as well)

What do you think?

Reply:
Si ha senso

Reply:
discussed offline, can't do this and also reusing the BRC type def :/